src/admin/index.js

```diff
@@ -190,6 +190,21 @@ module.exports = ({
         await cluster.refreshMetadataIfNecessary()
 
         const metadata = cluster.findTopicPartitionMetadata(topic)
+        if (fromTimestamp != null) {
+          const offsets = await cluster.fetchTopicsOffset([
+            {
+              topic,
+              fromTimestamp,
+              partitions: metadata.map(p => ({ partition: p.partitionId })),
+            },
+          ])
+          const { partitions } = offsets.pop()
```

Comment:
Is this safe? What happens if you fetch topic offsets with a timestamp for which there are no offsets (such as in the future, for example)?

Reply:
For a future timestamp, the result will be `offset: '-1'`, similar as `latest`. Maybe fallback to `high` was better?

Reply:
Now fallback to `higher offset` when got `-1`.

Reply:
Sorry for not getting back to you on this, but I thought you had abandoned the PR after finding the Kafka cli tools.

What I meant by my comment was whether `offsets` is guaranteed to be an array with at least one element? Otherwise this will throw.

Reply:
It's OK, I were just want to tell others about the command line and not expressed clearly.

For `offsets`, it is the same as `high` and `low` which called `.pop()` directly before this commit, and should contain one element.